src/NewChain/AuxiliaryChain.ts

```diff
@@ -0,0 +1,548 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import Web3 = require('web3');
+import * as RLP from 'rlp';
+import { ContractInteract } from '@openst/mosaic.js';
+
+import CliqueGenesis from './CliqueGenesis';
+import Contracts from './Contracts';
+import Shell from '../Shell';
+import Directory from '../Directory';
+import Logger from '../Logger';
+import NodeDescription from '../Node/NodeDescription';
+import GethNode from '../Node/GethNode';
+import { Tx } from 'web3/eth/types';
+import InitConfig from '../Config/InitConfig';
+import MosaicConfig from '../Config/MosaicConfig';
+import Proof from './Proof';
+
+/**
+ * The new auxiliary chain that shall be created.
+ */
+export default class AuxiliaryChain {
+  private web3: Web3;
+
+  private chainId: string;
+  private originChainId: string;
+  private chainDir: string;
+  private initConfig: InitConfig;
+  private nodeDescription: NodeDescription;
+
+  private sealer: string;
+  private deployer: string;
+
+  constructor(
+    initConfig: InitConfig,
+    chainId: string,
+    originChainId: string,
+    nodeDescription: NodeDescription,
+  ) {
+    this.initConfig = initConfig;
+    this.chainId = chainId;
+    this.originChainId = originChainId;
+    this.nodeDescription = nodeDescription;
+
+    this.chainDir = path.join(nodeDescription.mosaicDir, this.chainId);
+  }
+
+  /**
+   * @returns The Web3 instance of this chain.
+   */
+  public getWeb3(): Web3 {
+    return this.web3;
+  }
+
+  /**
+   * @returns The chain id of this chain as gotten from the node.
+   */
+  public getChainId(): string {
+    return this.chainId;
+  }
+
+  /**
+   * Generates two new accounts with an ethereum node and adds the addresses to the mosaic config
+   * as auxiliaryOriginalSealer and auxiliaryOriginalDeployer. These accounts will be used to run
+   * the sealer and deploy the contracts on auxiliary.
+   */
+  public generateAccounts(mosaicConfig: MosaicConfig): MosaicConfig {
+    if (fs.existsSync(path.join(this.chainDir, 'keystore'))) {
+      this.logWarn(
+        'keystore already exists; removing; has to start with fresh accounts to pre-calculate deployment addresses',
+      );
+      fs.removeSync(path.join(this.chainDir, 'keystore'));
+    }
+    this.logInfo('generating auxiliary address for sealer and deployer');
+
+    const args = [
+      'run',
+      '--rm',
+      '--volume', `${this.chainDir}:/chain_data`,
+      '--volume', `${this.nodeDescription.password}:/password.txt`,
+      'ethereum/client-go:v1.8.23',
+      'account',
+      'new',
+      '--password', '/password.txt',
+      '--datadir', '/chain_data',
+    ];
+
+    Shell.executeDockerCommand(args);
```

Comment:
Right, mosaic chains only assumes docker is available, not the individual nodes, etc. The nodes are run as docker containers.

Updated the Shell to throw in case of an error return.